### PR TITLE
Add related visual stories links

### PR DESCRIPTION
**Pull request by gestchild**
For #10884

links

Screenshot 2024-06-11 at 16 10 46

~We'll need the promo fields to be filled in in Prismic before putting this live~ - done
Screenshot 2024-06-12 at 14 19 02

### CODE REVIEW

File: content/webapp/pages/visual-stories/[visualStoryId].tsx [hunk 1]
```diff
@@ -18,28 +24,120 @@ import { SimplifiedServerData } from '@weco/common/server-data/types';
...
+  documentId,
+  visualStories,
+}: {
+  documentId: string;
+  visualStories: VisualStoriesDocument[];
+}): VisualStoriesDocument[] => {
+  return visualStories.filter(
+    // We want to remove any visual stories that satisfy any of the following:
+    // 1) is the same visualStory that is being displayed based on its id
+    // 2) is linked to the same exhibition or event as the visualStory being displayed (we infer it is the same visual story)
```
Comment (rcantin-w) [inline content/webapp/pages/visual-stories/[visualStoryId].tsx:53]: Could 2 happen without 1?

**Review by rcantin-w — APPROVED** [review]
LGTM! 🎉 
I do wonder if all the VS fetching/filtering/displaying logic could be moved to its own component somewhere, but it _is_ only used once, so at the same time it does not make the most sense with how we operate.
I used to work with pages using components in a `views/` folder (same level as `pages` and `components`) which allowed us to split pages into smaller pieces, which helped with readability when debugging. 
That's a different conversation though, it just made me think of it!

File: content/webapp/pages/visual-stories/[visualStoryId].tsx (hunk 1 above)
Comment (gestchild) [reply content/webapp/pages/visual-stories/[visualStoryId].tsx:53]: yes, but I think I've worded this badly. Visual stories can show up at three different urls depending whether they are linked to an exhibition and event or neither.

We need to remove a visual stories from the list if the id in the url is the visual story id in the list, e.g. https://wellcomecollection.org/visual-stories/ZU0PSRIAACkATr2l
Or if the id in the url is the id of an exhibition or event that is linked from the visual story in the list, e.g. 
https://wellcomecollection.org/exhibitions/ZZP8BxAAALeD00jo/visual-stories

I'll have a go at rephrasing the comments

Comment (rcantin-w) [reply content/webapp/pages/visual-stories/[visualStoryId].tsx:53]: Ohhh yes that makes sense, thank you!

**Pull request merged**